systemtest/src/main/java/io/strimzi/systemtest/Constants.java

```diff
@@ -138,7 +138,7 @@
     String PATH_TO_KAFKA_CONNECT_S2I_CONFIG = PATH_TO_PACKAGING_EXAMPLES + "/connect/kafka-connect-s2i.yaml";
     String PATH_TO_KAFKA_MIRROR_MAKER_2_METRICS_CONFIG = PATH_TO_PACKAGING_EXAMPLES + "/metrics/kafka-mirror-maker-2-metrics.yaml";
 
-    String METRICS_CONFIG_YAML_NAME = "metrics-config.yml";
+    String METRICS_CONFIG_JSON_NAME = "metrics-config.json";
```

Comment:
Same as with the examples ... I suspect in the system tests these should stay as YAML files.

Reply:
I had some confusion with the changes, Is only renaming this key to metrics-key.json sufficient. In system tests where the metric-config.yml is used, there is some yaml to json conversion going on.

Reply:
Users are using YAML -> in the custom resources and in the Config Maps. That should not change. Only the internal use of it is changed. If this is used for both internal and external, you would need to create two keys and update the STs.

Reply:
Okay. Now I get it. I was thinking that the for both external and internal thing we are going the json format. Thanks a lot for clearing my doubt.

Reply:
I think it is probably because of this line which uses the internal name as a name for the external ConfigMap field: https://github.com/strimzi/strimzi-kafka-operator/blob/50b22d2490aed3ca19c20d89f31cdb60b73a148f/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java#L774

I think you should use here some custom name and not `AbstractModel.ANCILLARY_CM_KEY_METRICS` and then use the custom name also in this config map here. Tat would make it much more clear what exactly this config map is and make the code more readable.

Reply:
Okay got it

Reply:
@ShubhamRwt Sorry, this was supposed to be as a comment to https://github.com/strimzi/strimzi-kafka-operator/pull/5072#discussion_r646905630 and not here.

Reply:
Okay np. I am making the changes

Reply:
The custom name would be corresponding to .yml right? Since the configmap is corresponding to user

Reply:
Yes